### DISCUSSION TITLE
Best Flight Controller for RL

### DISCUSSION THREAD

**Discussion by jonyMarino** [microsoft/AirSim#3636] [General]
As we know, AirSim is focused on AI.
The most common is to train a drone through several iterations.
For the iteration to be as fast as possible, there are two main factors at play: The ability to restart each iteration as quickly as possible (through a reset) and the running speed of the iteration. When we analyze it for external flight controllers (PX4, ardupilot, etc.), the ability to synchronize (lockstep) is a must for a realistic simulation.
I would like you to share your experiences/knowledge with external flight controllers for training AI here.
For example, with PX4, we have not found a way to do a reset other than ending the process and starting another (which takes a long time). Is it like that on ardupilot too? Others?
@rajat2004

**Comment by rajat2004** [comment]
AFAIK, Yeah, there's no way to reset ArduPilot in-flight other than close and restart the process

Edit: Actually, turns out there is a way. AP has a `reboot` command which only works when disarmed, so in the Mavproxy console, `disarm force` followed by `reboot` is like a hard reset. Should test it out, I think the reset implementation in AP needs to be looked at. Plus haven't yet made some scripts to control AP vehicles, high time that I did
